front/src/App.vue

```diff
@@ -0,0 +1,247 @@
+<template>
+    <div id="app">
+        <login v-on:login="onLogin($event)" v-if="!isConnected"></login>
+        <room-editor v-if="!roomEditorHidden" v-model:hidden="roomEditorHidden" v-on:create-room="createRoom($event)" v-on:join-room="joinRoom($event)"></room-editor>
+        <div class="app_vertical_layout">
+            <div class="app_top_horizontal_layout">
+                <room-list v-model:rooms="rooms" v-on:room-selected="displayRoom($event)" v-on:room-quit="quitRoom($event)"></room-list>
+                <div class="app_messages_list_wrapper">
+                    <activity-bar v-bind:current-room="selectedRoom" v-bind:is-writing="isWriting"></activity-bar>
+                    <message-list v-bind:messages="messages"></message-list>
+                </div>
+                <user-list v-bind:users="users"></user-list>
+            </div>
+            <div class="app_bottom_horizontal_layout">
+                <div class="app_room_editor_start" @click="roomEditorHidden = false">
+                    <div class="app_room_editor_plus"></div>
+                </div>
+                <message-editor v-model:content="currentTypedMessage" v-on:send-message="onMessageSend($event)" v-on:start-writing="isWriting = true" v-on:stop-writing="isWriting = false"></message-editor>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import {
+    ScratchyService,
+    Room,
+    Message,
+    User
+} from './service'
+import MessageEditor from './components/MessageEditor.vue';
+import MessageList from './components/MessageList.vue';
+import ActivityBar from './components/ActivityBar.vue';
+import RoomEditor from './components/RoomEditor.vue';
+import UserList from './components/UserList.vue';
+import RoomList from './components/RoomList.vue';
+import Login from './components/Login.vue';
```

Comment:
<!-- import components -->

Reply:
those are just the imports of App, self explanatory.